runtime/cpp/generators/all_known.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generator for emboss_arithmetic_all_known_generated.inc.
+
+# Maximum number of explicit arguments in the recursive overloads.  This script
+# will generate overloads for 1...OVERLOADS-1 arguments, plus a special overload
+# that handles >=OVERLOADS arguments using a variadic template.
+OVERLOADS = 64
+
+# Copyright header in the generated code complies with Google policies.
+print("""// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// GENERATED CODE.  DO NOT EDIT.  REGENERATE WITH
+// runtime/cpp/generators/all_known.py""")
```

Comment:
Can `__file__` be used here?

Reply:
`__file__` works if you happen to run:

```sh
python3 runtime/cpp/generators/all_known.py > runtime/cpp/emboss_arithmetic_all_known_generated.inc
```

But not if you run:

```sh
cd runtime/cpp
python3 generators/all_known.py > emboss_arithmetic_all_known_generated.inc
```